### PR TITLE
Fix #3450

### LINKED ISSUE #3450: Dovecot drops login attemps when limit of 100 concurrent imap sessions is reached
## Environment & Version

- Docker Compose version v2.29.2
- Version: 2024.06

## Description

Dovecot drops some login attempts.

## Replication Steps

Login with more than 100 mail clients over imap

## Observed behaviour
Login attempts are dropped after circumstances above. 

## Expected behaviour

Make it possible to override dovecot proxy.conf setting of front container.

## Logs

```
mailu-front[3011235]: Sep 11 11:50:10 master: Warning: service(imap-login): process_limit (100) reached, client connections are being dropped
```

## Workaround
copy proxy.conf template from container to host system:
```
cd /mailu
docker compose exec front cat /dovecot_conf/proxy.conf > proxy.conf
```

increase process_limit for service imap-login:
```
diff -u proxy.conf.orig proxy.conf
--- proxy.conf.orig     2024-09-11 15:12:29.898119459 +0200
+++ proxy.conf  2024-09-11 14:29:07.835583265 +0200
@@ -114,6 +114,7 @@
   inet_listener imap-webmail {
    port = 10143
   }
+  process_limit = 200
 }
 
 service pop3-login {
```
Here is my modification of docker-compose.yml to make proxy.conf modification persistent across container rebuilds.
```
# diff -u docker-compose.yml.orig docker-compose.yml
--- docker-compose.yml.orig     2024-09-11 15:18:22.271573663 +0200
+++ docker-compose.yml  2024-09-11 14:32:02.980379832 +0200
@@ -40,6 +40,7 @@
     volumes:
       - "/mailu/certs:/certs"
       - "/mailu/overrides/nginx:/overrides:ro"
+      - "/mailu/proxy.conf:/dovecot_conf/proxy.conf:ro"
     depends_on:
       - resolver
     dns:
```

and then restart front container:
```
docker compose restart front
```

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

bug-fix

## What does this PR do?

Ensure we can do more than 100 parallel sessions.
This will use more RAM.

### Related issue(s)
- close #3450

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

File: core/admin/start.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def test_DNS():
 
 cmdline = [
     "gunicorn",
-    "--threads", f"{os.cpu_count()}",
+    "--threads", f"{os.environ.get('CPU_COUNT', 1)}",
```
Comment (ghostwheel42) [inline core/admin/start.py:67]: No need for a f-string:
```suggestion
    "--threads", os.environ.get('CPU_COUNT', '1'),
```

File: core/base/libs/socrate/socrate/system.py [hunk 2]
```diff
@@ -141,6 +141,7 @@ def clean_env():
         for item in os.environ.get('TLS', ALL_PORTS).split(','):
             if item in PORTS_REQUIRING_TLS:
                 os.environ[f'TLS_{item}']='True'
+    os.environ['CPU_COUNT'] = str(os.cpu_count())
```
Comment (ghostwheel42) [inline core/base/libs/socrate/socrate/system.py:144]: Do we want this overrideable from the outside?
```suggestion
    if 'CPU_COUNT' not in os.environ:
        os.environ['CPU_COUNT'] = str(os.cpu_count())
```

**Comment by nextgens** [comment]
IMHO the real question is whether we are happy to drop the "high security mode" on the backend... and if not what kind of knob we want

File: core/nginx/dovecot/proxy.conf [hunk 3]
```diff
@@ -161,4 +176,11 @@ service submission-login {
...
     port = 10025
    }
+  service_count = 0
+  client_limit = 25000
+  process_min_avail = {{ CPU_COUNT }}
+  process_limit = {{ CPU_COUNT }}
+  vsz_limit = 256M
 }
+
+!include_try /overrides/dovecot.conf
```
Comment (Diman0) [inline core/nginx/dovecot/proxy.conf:186]: https://github.com/Mailu/Mailu/blob/341afeb57d266c1d990d1c41a6c57f0d2852117a/docs/faq.rst?plain=1#L245
Documentation (see above link) should be extended that you can drop dovecot.conf in overrides/nginx to override dovecot in front.

Comment (nextgens) [reply core/nginx/dovecot/proxy.conf:186]: I'm unsure how it interacts with the nginx conf where we say that all '*.conf' will be imported into nginx

Should we exclude it somehow or let it be and accept the warning messages that may result?

Comment (Diman0) [reply core/nginx/dovecot/proxy.conf:186]: Maybe we can say it must be in an extra folder?
../overrides/nginx/dovecot/dovecot.conf
And then change the inc
`!include_try /overrides/dovecot/dovecot.conf`

Comment (nextgens) [reply core/nginx/dovecot/proxy.conf:186]: That's 12e94b76

**Review by Diman0 — APPROVED** [review]
LGTM

**Comment by nazar-pc** [comment]
I'd appreciate if you can look at https://github.com/Mailu/Mailu/issues/3467, this may have caused a regression for me with dovecot override.

**Pull request merged**